pkg/datapath/linux/node_ids.go

```diff
@@ -221,7 +226,11 @@ func (n *linuxNodeHandler) unmapNodeID(ip string) error {
 	}
 	if id, exists := n.nodeIDsByIPs[ip]; exists {
 		delete(n.nodeIDsByIPs, ip)
-		delete(n.nodeIPsByIDs, id)
+
+		n.nodeIPsByIDs[id].Delete(ip)
```

Comment:
for the sake of completeness: should we add a nil-check for `n.nodeIPsByIDs[id]` using the value (Set) ? :thinking: 

even though it's ensured by the (current) implementation that `nodeIPsByIDs` contains the entry - it kind of feels/reads wrong not having the check.

Reply:
Calling `Delete` on a nil Set is actually safe because it translates into a map key deletion under the hood, which is a no-op if the map is nil:

```
// The delete built-in function deletes the element with the specified key
// (m[key]) from the map. If m is nil or there is no such element, delete
// is a no-op.
func delete(m map[Type]Type1, key Type)
```